python_modules/dagster/dagster/_core/definitions/cacheable_assets.py

```diff
@@ -63,9 +64,9 @@ def __new__(
     ):
         extra_metadata = check.opt_nullable_mapping_param(extra_metadata, "extra_metadata")
         try:
-            # check that the value is JSON serializable
-            seven.dumps(extra_metadata)
-        except TypeError:
+            # check that the value can pass through the serdes layer
+            serialize_value(extra_metadata)
+        except SerializationError:
```

Comment:
PowerBI objects were failing serialization without this change-- `@record` objects aren't iterable and so `json.dumps` chokes on them, they need to undergo the preprocessing in `serialize_value`.